engine/include/cubos/engine/collisions/raycast.hpp

```diff
@@ -0,0 +1,255 @@
+/// @file
+/// @brief Component @ref cubos::engine::Raycast.
+/// @ingroup collisions-plugin
+
+#pragma once
+
+#include <utility>
+
+#include <glm/gtc/matrix_transform.hpp>
+#include <glm/vec3.hpp>
+
+#include <cubos/core/ecs/system/fetcher.hpp>
+#include <cubos/core/geom/box.hpp>
+
+#include <cubos/engine/collisions/collider.hpp>
+#include <cubos/engine/collisions/shapes/box.hpp>
+#include <cubos/engine/collisions/shapes/capsule.hpp>
+#include <cubos/engine/prelude.hpp>
+#include <cubos/engine/transform/local_to_world.hpp>
+#include <cubos/engine/transform/position.hpp>
+
+namespace cubos::engine
+{
+    struct Hit
+    {
+        Entity entity;
+        glm::vec3 point;
+    };
+
+    struct Ray
+    {
+        glm::vec3 origin, direction;
+    };
+
+    class Raycast
```

Comment:
Missing docs. It would be nice to specify that it is a system argument, and have a code snippet explaining how to use it.